solutions/swb-app/src/userRoutes.ts

```diff
@@ -151,20 +161,24 @@ export function setUpUserRoutes(router: Router, userService: UserManagementServi
   router.post(
     '/roles',
     wrapAsync(async (req: Request, res: Response) => {
-      processValidatorResult(validate(req.body, CreateRoleSchema));
-      const response = await userService.createRole(req.body.roleName);
+      const validatedRequest = validateAndParse<CreateRoleRequest>(CreateRoleRequestParser, req.body);
+      const response = await userService.createRole(validatedRequest.roleName);
       res.status(201).send(response);
     })
   );
 
   router.put(
     '/roles/:roleName',
     wrapAsync(async (req: Request, res: Response) => {
-      processValidatorResult(validate(req.body, UpdateRoleSchema));
-      if (!_.isString(req.params.roleName)) {
+      const validatedRequest = validateAndParse<UpdateRoleRequest>(UpdateRoleRequestParser, {
+        ...req.body,
+        roleName: req.params.roleName
+      });
+
+      if (!_.isString(validatedRequest.roleName)) {
         throw Boom.badRequest('roleName must be a string.');
       }
-      const response = await userService.addUserToRole(req.body.username, req.params.roleName);
+      const response = await userService.addUserToRole(validatedRequest.username, validatedRequest.roleName);
       res.send(response);
     })
```

Comment:
Can you create a backlog item to remove the Role APIs? These shouldn't have been left in here.

Reply:
created item GALI-2373